lib/packages/authentication/bloc/authentication_bloc.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:async';
+
+import 'package:coffee_app/packages/authenticationRepository/authentication_repository.dart';
+import 'package:coffee_app/packages/userRepository/user_repository.dart';
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import '../../userRepository/models/user.dart';
+
+part 'authentication_event.dart';
+part 'authentication_state.dart';
+
+class AuthenticationBloc
+    extends Bloc<AuthenticationEvent, AuthenticationState> {
+  AuthenticationBloc({
+    required AuthenticationRepository authenticationRepository,
+    required UserRepository userRepository,
+  })  : _authenticationRepository = authenticationRepository,
+        _userRepository = userRepository,
+        super(const AuthenticationState.unknown()) {
+    on<AuthenticationSubscriptionRequested>(_onSubscriptionRequested);
```

Comment:
Don't register multiple `on`, use only one and handle all subevents inside. Please, use the following as a reference.
https://github.com/hawkkiller/sizzle_starter/blob/main/lib/src/feature/settings/bloc/app_settings_bloc.dart